### ISSUE TITLE
Page numbers for each page

### ISSUE THREAD

**Issue by Higgs32584** [ankushshah89/python-docx2txt#44]
Hi, so there is not a simple way to tell what page number some text is from.Could we add functionality to divide text by page numbers? Thank you

**Comment by aronweiler** [comment]
I too, am wondering how to figure out what page things are on... 

Is there a way to split docs on page breaks?  Or a way to somehow add something to the metadata that includes the pages?

**Comment by aronweiler** [comment]
> Not possible without rebuilding the Word layout engine.

Yeah, I started poking through the specifications and format for the XML underlying the Word documents, and it seems like page numbers are a completely client-side thing.  

There are possible work-arounds, like using the last rendered page break, but that is reliant on someone opening a word document (in Word), scrolling through it, and then re-saving it as Word itself adds the `<w:lastRenderedPageBreak/>` XML fragment when it is rendering.

Obviously that's a bad idea to rely on for several reasons, and one appears to be that it doesn't even put the XML fragments in the right place all of the time.  

For some context around what I wanted page numbers for- I am using [docx2text](https://github.com/ankushshah89/python-docx2txt) to load word documents into a vector store to be consumed by an LLM, and I wanted to print out source locations inside of the documents that I found data in.

**Comment by phanikavi** [comment]
@aronweiler  - I have the same requirement. Did you figure out a way? Thanks in advance.

> I wanted page numbers for- I am using [docx2text](https://github.com/ankushshah89/python-docx2txt) to load word documents into a vector store to be consumed by an LLM, and I wanted to print out source locations inside of the documents that I found data in.

**Comment by ShayHill** [comment]
It isn’t possible. Page numbers are not included in the xml, because they are only calculated after the xml is rendered. I have a similar library, and that is the most requested feature.

**Comment by aronweiler** [comment]
@phanikavi like @ShayHill said, it's not possible... Ultimately I moved on to other libraries, usually just converting to PDF first.

**Comment by ShayHill** [comment]
The real solution may be to get your content from docx, where you have better context, then align it to page numbers in a Word-generated pdf. I offered to do that for a client. They declined, because they didn’t want to maintain pdf copies and the formatting in their docx files wasn’t consistent. Non substantive changes like formatting would have changed the output. So I never did see how straightforward it would be to automate docx to pdf conversion *exactly* how Word would do it.

But their project was mostly tables, and most of what they wanted were the values in those tables. References to page numbers in a theoretical rendering of their thousands of docx files were a nice to have they decided they could live without.
